src/components/structures/RoomView.js

```diff
@@ -812,10 +812,10 @@ export default createReactClass({
         /* Check all verified user devices. */
         for (const userId of verified) {
             const devices = await cli.getStoredDevicesForUser(userId);
-            const allDevicesVerified = devices.every(({deviceId}) => {
-                return cli.checkDeviceTrust(userId, deviceId).isVerified();
+            const anyDeviceNotVerified = devices.some(({deviceId}) => {
+                return !cli.checkDeviceTrust(userId, deviceId).isVerified();
             });
-            if (!allDevicesVerified) {
+            if (anyDeviceNotVerified) {
```

Comment:
What's this change for? Isn't this a negation of the same predicate which is then be re-negated in the if clause?

Reply:
Its just a bail-early optimization, the old code checked if every item was X, this code checks if any item is !X

Reply:
ah, of course